### PR TITLE
feat(appellate): Reference district cases from Appellate

### PR DESCRIPTION
**Pull request by ERosendo**
this PR resolves https://github.com/freelawproject/recap/issues/21.

This PR includes the following changes:

- This PR adds two new helper methods to make **core docket numbers**. The first helper converts en & em dashes to hyphens, and it's called `normalizeDashes`, the other method makes a **core docket number** from an existing docket number, and it's called `makeDocketNumberCore`. This PR also adds tests for these methods.

- A new parameter is added to the getAvailabilityForDocket method. This new parameter is called core_docket_number and allows us to filter dockets using the `pacer_case_id `or the `docket_number_core`.

- This PR removes the `findAndStorePacerDocIds` method from the ContentDelegate class constructor. This method is asynchronous, so we should wait until it finds the doc links on the page, but we cannot use the await operator inside a constructor. This PR also updates all the invocations of the `getAvailabilityForDocket` in this class.

- Three helper functions are added to the Appellate utils.
    - `getTableWithDataFromCaseSelection`: We move some code from the `getCaseIdFromCaseSelection` method into this helper function. This function returns the table with data from the **Case Selection Page** if it's available.
    - `getDatafromDistrictLinkUrl`: this helper extracts the **court id** and the **docket number** from links to the district court available on the Case Selection Page. This utility function also computes the **core docket number**.
    - `makeRButtonForCases`: This function creates a div element that contains an anchor with the link to the district case in the RECAP archive and the RECAP icon.

- This PR moves some checks from the function that handles the response of the `getAvailabilityForDocket` method into a new helper called `handleDocketAvailabilityMessages`.

- This PR updates the `handleCaseSelectionPage` method from the `AppellateDelegate` to inject [R] icons next to the two links on the page. One [R] for the appellate case, and one for the district case ( I used the approach you suggested [here](https://github.com/freelawproject/recap-chrome/pull/296#issuecomment-1366848529)). Here's a gif with that shows the new feature:

Appellate-district-link

### Additional comments

- The response from the `getAvailabilityForDocket` might contain more than one result, in such cases the extension doesn't insert the [R] icon. Here's a gif showing this case:

Appellate-district-link-error

### CODE REVIEW

File: spec/RecapSpec.js
Comment (mlissner) [inline spec/RecapSpec.js:104]: Nit. It's a bit sloppy how we do the ampersand here. It'd be better to always have it at the beginning:

```
'https...' +
'?xyz=foo' +
'&y=bar' +
'&z=bas';
```

File: src/appellate/appellate.js
Comment (mlissner) [inline src/appellate/appellate.js:93]: I guess we should add prettier to our CI system somehow, but shouldn't this be:

```suggestion
        } else {
```

To be clear, I don't really care about this one instance, but it'd be good to be confident that our code is always clean and tidy. Some things prettier does are more important too, like adding semicolons?

File: src/appellate/utils.js
Comment (mlissner) [inline src/appellate/utils.js:266]: Helpful to have an example URL here.

**Review by mlissner — CHANGES_REQUESTED** [review]
No major comments here. In general looks great and if we wait until April to merge this, it'll fix a bug that's ten years old!

Thanks.

File: spec/RecapSpec.js
Comment (ERosendo) [reply spec/RecapSpec.js]: You're right! 

Changed. Thanks!

File: src/appellate/appellate.js
Comment (ERosendo) [reply src/appellate/appellate.js]: You're right about the styling here.

File: src/appellate/utils.js
Comment (ERosendo) [reply src/appellate/utils.js:266]: We have one! I'm going to add an example. Thanks

**Comment by ERosendo** [comment]
Thanks for your comments, I've added the changes you requested.

**Review by mlissner — APPROVED** [review]

**Pull request merged**
